lxd/project/permissions.go

```diff
@@ -1174,3 +1176,31 @@ func FilterUsedBy(r *http.Request, entries []string) []string {
 
 	return usedBy
 }
+
+// CheckClusterTargetRestriction check if user is allowed to use '--target' flag
+func CheckClusterTargetRestriction(tx *db.ClusterTx, r *http.Request, projectName string, targetFlag string) error {
+	if rbac.UserIsAdmin(r) {
+		return nil
+	}
+
+	project, err := tx.GetProject(projectName)
+	if err != nil {
+		return fmt.Errorf("Fetch project database object")
+	}
+
+	if !projectHasLimitsOrRestrictions(project) {
+		return nil
+	}
+
+	key := "restricted.cluster.target"
+	restrictionValue, ok := project.Config[key]
+	if !ok {
+		restrictionValue = defaultRestrictionsValues[key]
+	}
+
+	if restrictionValue == "block" && targetFlag != "" {
+		return fmt.Errorf("Use of '--tagret' flag is prohibited when restricted.cluster.target key is set to 'block'")
```

Comment:
Typo in there but also, we can't assume that the user is using our command line tool, so you'd want something more generic, like:

`This project doesn't allow cluster member targeting`